app/components/autocomplete/slash_suggestion/slash_suggestion.js

```diff
@@ -96,16 +94,31 @@ export default class SlashSuggestion extends PureComponent {
         this.props.onResultCountChange(data.length);
     }
 
-    filterSlashSuggestions = (matchTerm, commands) => {
-        return commands.filter((command) => {
-            if (!command.auto_complete) {
-                return false;
-            } else if (!matchTerm) {
-                return true;
+    filterSlashSuggestions = (matchTerm, commands, dynamicArguments) => {
+        const autocompleteData = commands.map((command) => {
+            if (command.autocomplete_data) {
+                return command.autocomplete_data;
             }
-
-            return command.display_name.startsWith(matchTerm) || command.trigger.startsWith(matchTerm);
+            return {
+                Trigger: command.trigger,
+                HelpText: command.auto_complete_desc,
+                Hint: command.auto_complete_hint,
+            };
         });
+        const commandAutocomplete = new CommandAutocomplete(this.updateDynamicAutocompleteSuggestions);
+        return commandAutocomplete.getSuggestions(autocompleteData, dynamicArguments, '', matchTerm);
+    }
+
+    updateDynamicAutocompleteSuggestions = (url, parsed, toBeParsed) => {
+        if (!this.state.dynamicArgumentsLastFetch[url] ||
+            Date.now() - this.state.dynamicArgumentsLastFetch[url] > TIME_BEFORE_NEXT_DYNAMIC_ARGUMENTS_REQUEST) {
+            this.props.actions.getDynamicAutocompleteSuggestions(url, parsed, toBeParsed, this.props.currentTeamId);
```

Comment:
This is being called here regardless of the server version, if the server  does not support this, we are triggering an action which the request is going to fail